Bonsai32/Bonsai32.csproj

```diff
@@ -1,21 +1,34 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <Title>Bonsai32</Title>
     <Description>The x86 bootstrapper for the Bonsai environment.</Description>
     <PackageTags>Bonsai Rx Reactive Extensions</PackageTags>
     <UseWindowsForms>true</UseWindowsForms>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
-    <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
+    <IsPackable>false</IsPackable>
     <TargetFramework>net48</TargetFramework>
     <PlatformTarget>x86</PlatformTarget>
-    <VersionPrefix>2.8.3</VersionPrefix>
     <OutputType>Exe</OutputType>
-    <OutputPath>..\Bonsai\bin\$(Configuration)\</OutputPath>
+    <OutputPath>$(ArtifactsPath)\bin\Bonsai\$(Configuration.ToLowerInvariant())\</OutputPath>
     <ApplicationIcon>..\Bonsai.Editor\Bonsai.ico</ApplicationIcon>
     <ApplicationManifest>App.manifest</ApplicationManifest>
+
+    <!--
+      These along with the `Private="false"` on the ProjectReference below ensure that MSBuild does not attempt to copy
+      the main Bonsai.exe and its dependencies to the output of Bonsai32 since it will be output to the main bootstrapper's
+      output folder (which already has all of that.)
+    -->
+    <CopyLocalLockFileAssemblies>false</CopyLocalLockFileAssemblies>
+    <DisableTransitiveProjectReferences>true</DisableTransitiveProjectReferences>
   </PropertyGroup>
   <ItemGroup>
-    <ProjectReference Include="..\Bonsai\Bonsai.csproj" />
+    <ProjectReference Include="..\Bonsai\Bonsai.csproj" Private="false" />
+
+    <!--
+      This reference is necessary to work around a ambiguity in the NuGet restore graph.
```

Comment:
```suggestion
      This reference is necessary to work around an ambiguity in the NuGet restore graph.
```

Interesting, do you have a reference for this? So technically this shouldn't add a dependency to the 32-bit bootstrapper, right? It still only calls the main bootstrapper in 32-bit mode which then resolves everything from its own internal logic.

Reply:
Yeah, this is a pure restore-time issue.

No reference, this is a very specific weird case caused by how Bonsai is structured.

I don't remember all the details (this is like one of the first things I did on this PR), but basically Bonsai32 can "see" an indirect dependency to `System.Runtime.CompilerServices.Unsafe/4.5.3` in the restore graph via the path `Bonsai32` -> `Bonsai` -> `Bonsai.Design` -> `System.Resources.Extensions/8.0.0` -> `System.Memory/4.5.5` -> `System.Runtime.CompilerServices.Unsafe/4.5.3`

![Bonsai32 restore graph visualization](https://github.com/bonsai-rx/bonsai/assets/278957/0b68301c-579f-4307-9cdc-3398cd223018)

Meanwhile there exists within Bonsai an indirect dependency to `System.Runtime.CompilerServices.Unsafe/6.0.0` via the path `Bonsai` -> `System.Reflection.MetadataLoadContext/7.0.0` -> `System.Collections.Immutable/7.0.0` -> `System.Runtime.CompilerServices.Unsafe/6.0.0` as well as the similar path to `System.Runtime.CompilerServices.Unsafe/4.5.3` through `System.Memory/4.5.5` as seen by Bonsai32:

![Bonsai restore graph visualization](https://github.com/bonsai-rx/bonsai/assets/278957/bdeb59de-1b70-448f-859a-6321058d9fa8)

In the case of Bonsai it resolves to 6.0.0 since it's the newest dependency and satisfies every package's constraints.

However Bonsai32 does not see the indirect 6.0.0 dependency in its restore graph because Bonsai marks `System.Reflection.MetadataLoadContext` as private.

NuGet doesn't handle this situation properly (I think it catches it too late) and you get MSB3277 as a result. (The comment is maybe a bit wordy since this is unlikely to happen again, but my hope was I could save someone some time in case it somehow did.)